python/ray/dataframe/dataframe.py

```diff
@@ -1419,9 +1462,24 @@ def duplicated(self, subset=None, keep='first'):
             "github.com/ray-project/ray.")
 
     def eq(self, other, axis='columns', level=None):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Checks element-wise that this is equal to other.
+
+        Args:
+            other: A DataFrame or Series or scalar to compare to.
+            axis: The axis to perform the eq over.
+            level: The Multilevel index level to apply eq over.
+
+        Returns:
+            A new DataFrame filled with Booleans.
+        """
+        if isinstance(other, DataFrame):
+            return self._inter_df_op_helper(
+                lambda x, y: x.eq(y, axis, level),
+                other, axis, level)
+        else:
+            return self._single_df_op_helper(
+                lambda df: df.eq(other, axis, level),
+                other, axis, level)
```

Comment:
instead of this `if` statement, should we just use `_iter_and_single_df_op_helper`?